server/src/main/java/org/elasticsearch/action/search/CanMatchNodeRequest.java

```diff
@@ -146,6 +152,40 @@ public CanMatchNodeRequest(
         indices = shards.stream().map(Shard::getOriginalIndices).flatMap(Arrays::stream).distinct().toArray(String[]::new);
     }
 
+    private static void collectSignificantTermsBackgroundFilters(
+        Collection<AggregationBuilder> aggregations,
+        List<QueryBuilder> backgroundFilters
+    ) {
+        for (AggregationBuilder aggregation : aggregations) {
+            if (aggregation instanceof SignificantTermsAggregationBuilder) {
```

Comment:
would it make sense to add a new default method to AggregationBuilder to avoid the instanceof check?

Reply:
Good suggestion, thanks. I've updated the PR accordingly.